R/create.R

```diff
@@ -843,3 +843,82 @@ create_stan_params <- function(params) {
   }
   c(ret, as.list(ids), as.list(null_ids))
 }
+
+#' Create Summary Output from Infection Estimation Objects
+#'
+#' @description `r lifecycle::badge("stable")`
+#'
+#' This function creates summary output from infection estimation objects
+#' (either `estimate_infections` or `forecast_infections`). It is used
+#' internally by [summary.estimate_infections()] and
+#' [summary.forecast_infections()] to provide a consistent summary interface.
+#'
+#' @param object An infection estimation object (either from
+#'   [estimate_infections()] or [forecast_infections()]).
+#'
+#' @param type A character vector of data types to return. Defaults to
+#'   "snapshot" but also supports "parameters". "snapshot" returns
+#'   a summary at a given date (by default the latest date informed by data).
+#'   "parameters" returns summarised parameter estimates that can be further
+#'   filtered using `params` to show just the parameters of interest and date.
+#'
+#' @param date A date in the form "yyyy-mm-dd" to inspect estimates for.
```

Comment:
If it actually is a date object does the format matter? or is it a character string to be coerced?